### DISCUSSION TITLE
Seeking community guidance for the future of the project

### DISCUSSION THREAD

**Discussion by machulav** [machulav/ec2-github-runner#172] [General]
Dear community,

First and foremost, I want to extend my deepest gratitude to each of you for your invaluable support and contributions to the project.

As the creator of ec2-github-runner, it has been a privilege and an incredible journey to witness and lead its growth. However, I have reached a point where my availability to dedicate time and resources to this project is becoming increasingly limited. My commitment to maintaining the quality and responsiveness that this project deserves is unwavering, but I must acknowledge my time constraints.

Therefore, I am reaching out to you for guidance and suggestions on the best way forward. There are several paths we could consider:

1. **Seeking Maintainers:** If some of you are willing and able to step up as maintainers, it would be a great way to ensure the continued development and support of ec2-github-runner.

2. **Community-Driven Development:** Transforming the project into a more community-driven model where decisions and developments are more collaborative. However, the question is how to ensure the quality and stability of releases, as every release requires a lot of manual testing because of the specifics of two platforms: AWS EC2 and GitHub Actions.

3. **Archiving the Project:** If finding new maintainers is not feasible, we might consider archiving the project, ensuring that the current state remains accessible for reference and use.

4. **Your Suggestions:** I am open to any other ideas or proposals you might have.

This project is as much yours as it is mine, and its future should reflect the wishes and capabilities of this community. Please share your thoughts, willingness to contribute, and any other suggestions you might have. Your input is crucial in shaping the next chapter of ec2-github-runner.

Thank you once again for your remarkable support. I look forward to your insights and hope for a path that best serves our community and the project's legacy.

Warm regards,
Volodymyr

**Comment by Dmitry1987** [comment]
thanks for all your work @machulav 🙏  
I can help maintain the project, it's a very useful one and I don't really know why aws or github do not have this feature as a built-in integration in them (while millions of users use both gh and aws).

About the testing, isn't all testing automated? the verification of stable release seems simple - run a small instance and see if it got connected to a repo as a fresh runner, execute a few commands on it to validate that any inputs which were passed are actually applied (like disk size or EBS mapping are actually the correct size etc'). 

The solution should probably be using option 2, community-driven development with a help of primary lead maintainers who actually cut the releases after they did the testing. I can try to be one of the few maintainers who will ensure testing.

**Comment by Preen** [comment]
I will gladly help out in any way possible.

**Comment by srilman** [comment]
@machulav This is a great project that really bridges the gap between using Github-hosted runners and all of the complex self-hosted solutions available. But as of January 2024, it does feel like this repo has started to become stale, with the last commit from 6 months ago. Furthermore, I've noticed that there are many forks containing highly-requested features, like spot instances and Windows support.

I'd suggest that you seek another maintainer (option 1) for the time being to help with the backlog of feature PRs that need to be tested so they can be merged soon and a new version can be released. At that point, we can revisit this conversation. In the long term, it would make sense to lean towards option 2, but we should find a way to automate some of the testing beforehand.

**Comment by ctfaria2** [comment]
Hi @machulav. I am happy to help out as a maintainer.

**Comment by machulav** [comment]
Please contact me by email at arcs-lipid0t@icloud.com to arrange the contributions.

**Comment by Preen** [comment]
There are breaking changes that needs supervising that could break this plugin that needs supervision asap. There are already fixes for these in pull requests.

There's two ways of handling this, either you need to let people in that can help out with checking pull requests out and approve them or we need to branch off and setup another repo.

```bash
(node:1691) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
Please migrate your code to use AWS SDK for JavaScript (v3).
```

**Comment by machulav** [comment]
We have a new maintainer for the project! See the announcement in #202. 
For this reason, I’m closing this conversation.
